### PR TITLE
feat(Linux): switch supported architectures to GA Temurin jdk21 images

### PR DESCRIPTION
**Pull request by lemeurherve**
Follow-up of https://github.com/jenkinsci/docker-agent/pull/521

Needs #437 and the subsequent updatecli PR updating parent image version in Dockerfiles.

EDIT: cherry-picked corresponding changes.

### Testing done

https://ci.jenkins.io/job/Packaging/job/docker-inbound-agent/job/PR-436/13/

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by lemeurherve** [comment]
Cherry-picked https://github.com/jenkinsci/docker-inbound-agent/pull/438

File: debian/preview/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG version=3180.v3dd999d24861-1
+ARG version=3180.v3dd999d24861-2
```
Comment (lemeurherve) [inline debian/preview/Dockerfile:1]:
```suggestion
ARG version=3180.v3dd999d24861-2-preview
```

**Comment by lemeurherve** [comment]
I had to revert https://github.com/jenkinsci/docker-inbound-agent/pull/436/commits/055ae83ee3541d0281c4dced0b96f9130342f6fc as the jdk21-preview parent image doesn't include these architectures: https://hub.docker.com/layers/jenkins/agent/3180.v3dd999d24861-2-jdk21-preview/images/sha256-bede7232308fb7fe5432bc5cf72a58623676b3e01471f5ff965ef1dd3e8109ff?context=explore

To build this debian_jdk_21_preview image in builds, more modifications to the Makefile are required to include architectures other than linux/amd64 & linux/arm64. Something like we've already discussed with @dduportal, building each Linux variant on its own agent with corresponding architecture. (somehow like what's done for Windows images)

**Comment by lemeurherve** [comment]
Another easier (?) solution while we're keeping preview images with EA Temuring binaries would be to trigger another docker-agent release with its debian_jdk21_preview target also containing linux/md64 & linux/arm64 architectures.

**Comment by lemeurherve** [comment]
Updatecli is failing because of https://github.com/jenkinsci/docker-inbound-agent/pull/436/commits/54b2e1204cd5075278212f76aef1333900964b63, will be restored when merged.

**Review by dduportal — APPROVED** [review]
LGTM!

**Comment by dduportal** [comment]
> I had to revert [055ae83](https://github.com/jenkinsci/docker-inbound-agent/commit/055ae83ee3541d0281c4dced0b96f9130342f6fc) as the jdk21-preview parent image doesn't include these architectures: https://hub.docker.com/layers/jenkins/agent/3180.v3dd999d24861-2-jdk21-preview/images/sha256-bede7232308fb7fe5432bc5cf72a58623676b3e01471f5ff965ef1dd3e8109ff?context=explore, which has for immediate result the fact the debian_jdk_21_image target isn't built anymore in this PR.
> 
> To build this debian_jdk_21_preview image in builds, more modifications to the Makefile are required to include architectures other than linux/amd64 & linux/arm64. Something like we've already discussed with @dduportal, building each Linux variant on its own agent with corresponding architecture. (somehow like what's done for Windows images)

I'm not sure to understand what is the need to change Makefile?

As you underlined, the problem come from the parent repository not publishing the `linux/amd64` and `linux/arm64` 21 preview images right?

**Comment by lemeurherve** [comment]
> > I had to revert [055ae83](https://github.com/jenkinsci/docker-inbound-agent/commit/055ae83ee3541d0281c4dced0b96f9130342f6fc) as the jdk21-preview parent image doesn't include these architectures: https://hub.docker.com/layers/jenkins/agent/3180.v3dd999d24861-2-jdk21-preview/images/sha256-bede7232308fb7fe5432bc5cf72a58623676b3e01471f5ff965ef1dd3e8109ff?context=explore, which has for immediate result the fact the debian_jdk_21_image target isn't built anymore in this PR.
> > To build this debian_jdk_21_preview image in builds, more modifications to the Makefile are required to include architectures other than linux/amd64 & linux/arm64. Something like we've already discussed with @dduportal, building each Linux variant on its own agent with corresponding architecture. (somehow like what's done for Windows images)
> 
> I'm not sure to understand what is the need to change Makefile?
> 
> As you underlined, the problem come from the parent repository not publishing the `linux/amd64` and `linux/arm64` 21 preview images right?

In the Makefile the list of images to build is retrieved by https://github.com/jenkinsci/docker-inbound-agent/blob/master/Makefile#L47-L48

This rule excludes the preview image. The easiest solution to build it is to reintroduce amd64/arm64 architectures in debian_jdk_21_preview image of docker-agent, the alternative would be to change this listing (maybe?)

**Comment by dduportal** [comment]
> > > I had to revert [055ae83](https://github.com/jenkinsci/docker-inbound-agent/commit/055ae83ee3541d0281c4dced0b96f9130342f6fc) as the jdk21-preview parent image doesn't include these architectures: https://hub.docker.com/layers/jenkins/agent/3180.v3dd999d24861-2-jdk21-preview/images/sha256-bede7232308fb7fe5432bc5cf72a58623676b3e01471f5ff965ef1dd3e8109ff?context=explore, which has for immediate result the fact the debian_jdk_21_image target isn't built anymore in this PR.
> > > To build this debian_jdk_21_preview image in builds, more modifications to the Makefile are required to include architectures other than linux/amd64 & linux/arm64. Something like we've already discussed with @dduportal, building each Linux variant on its own agent with corresponding architecture. (somehow like what's done for Windows images)
> > 
> > 
> > I'm not sure to understand what is the need to change Makefile?
> > As you underlined, the problem come from the parent repository not publishing the `linux/amd64` and `linux/arm64` 21 preview images right?
> 
> In the Makefile the list of images to build is retrieved by https://github.com/jenkinsci/docker-inbound-agent/blob/master/Makefile#L47-L48
> 
> This rule excludes the preview image. The easiest solution to build it is to reintroduce amd64/arm64 architectures in debian_jdk_21_preview image of docker-agent, the alternative would be to change this listing (maybe?)

As discussed directly:

- It looks like the docker bake's group `linux` does NOT have the 21_preview images. This group is used on ci.jenkins.io for the multi-arch "full bake build" (after the linux/amd64 are built and tested by Make) and is used on trusted.ci.jenkins.io for the real-life "build and deploy". It means the latest release might not be published for 21 preview (s390x/ppc/arm32) => fix is required on both jenkinsci/docker-agent and here with new releases.
- It is absolutely acceptable to define a set of `linux-<arch>` including amd64 and arm64, and slightly update the Makefile to use them instead of the global `linux`
- Long term: I propose that we simplify the equation and see if we can either remove of delete the `Makefile` in favor of docker buildx bake so that ci.jenkins.io would execute almost the same command as trusted.ci.jenkins.io

**Pull request merged**